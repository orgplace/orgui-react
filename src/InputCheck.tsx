import classNames from 'classnames';
import React from 'react';

interface OuiInputCheckAttributes {
  checked?: boolean;
  onChange?: React.ChangeEventHandler<HTMLInputElement>;
}

const OuiInputCheck: React.FC<React.HTMLAttributes<HTMLElement> & OuiInputCheckAttributes> = (props) => {
  const {checked, onChange, children, ...labelProps} = props;
  return (
    <label {...labelProps} className={classNames(
      'ouiInputCheck',
      props.className,
    )}>
      <input type="checkbox" className="ouiInputCheck__input" checked={checked} onChange={onChange} />
      <span className="ouiInputCheck__text">{children}</span>
    </label>
  );
}

export { OuiInputCheck };
