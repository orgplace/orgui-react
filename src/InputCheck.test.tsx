import React from 'react'
import { OuiInputCheck } from './InputCheck'
import renderer from 'react-test-renderer';


test('OuiInputCheck', function() {
	const component = renderer.create(<OuiInputCheck />);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
