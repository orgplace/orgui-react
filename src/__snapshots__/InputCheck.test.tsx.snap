// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OuiInputCheck 1`] = `
<label
  className="ouiInputCheck"
>
  <input
    className="ouiInputCheck__input"
    type="checkbox"
  />
  <span
    className="ouiInputCheck__text"
  />
</label>
`;
